test/gtest/api_convbiasactiv.cpp

```diff
@@ -0,0 +1,188 @@
+    friend std::ostream& operator<<(std::ostream& os, const CBATestCase& tc)
+    {
+        return os << "N: " << tc.N << " C:" << tc.C << " H:" << tc.H << " W:" << tc.W
+                  << " k: " << tc.k << " y:" << tc.y << " x:" << tc.x << " pad_y:" << tc.pad_y
+                  << " pad_x:" << tc.pad_x << " stride_y:" << tc.stride_y
+                  << " dilation_y:" << tc.dilation_y << " activ_mode:" << tc.activ_mode
+                  << " conv_mode:" << tc.conv_mode;
+    }
+};
+
+struct ConvBiasActivFwdTest
+    : public ::testing::TestWithParam<std::tuple<miopenConvFwdAlgorithm_t, CBATestCase>>
+{
+protected:
+    void SetUp() override
+    {
+        std::tie(algo, cba_config) = GetParam();
+        const double double_zero   = 0.0f;
+        input   = tensor<float>{cba_config.N, cba_config.C, cba_config.H, cba_config.W};
+        weights = tensor<float>{1, cba_config.k, cba_config.x, cba_config.y};
+        input.generate(tensor_elem_gen_integer{17});
+        weights.generate(tensor_elem_gen_integer{17});
+        miopenCreateConvolutionDescriptor(&conv_desc);
+        miopenCreateActivationDescriptor(&activ_desc);
+        miopenSetActivationDescriptor(
+            activ_desc, cba_config.activ_mode, double_zero, double_zero, double_zero);
+        miopenInitConvolutionDescriptor(conv_desc,
+                                        cba_config.conv_mode,
+                                        cba_config.pad_y,
+                                        cba_config.pad_x,
+                                        cba_config.stride_y,
+                                        cba_config.stride_x,
+                                        cba_config.dilation_y,
+                                        cba_config.dialtion_x);
+        int n, c, h, w;
+        miopenGetConvolutionForwardOutputDim(conv_desc, &input.desc, &weights.desc, &n, &c, &h, &w);
+        output  = tensor<float>{static_cast<size_t>(n),
+                               static_cast<size_t>(c),
+                               static_cast<size_t>(h),
+                               static_cast<size_t>(w)};
+        ref_out = tensor<float>{static_cast<size_t>(n),
+                                static_cast<size_t>(c),
+                                static_cast<size_t>(h),
+                                static_cast<size_t>(w)};
+        bias    = tensor<float>{1, static_cast<size_t>(c), 1, 1};
+        bias.generate(tensor_elem_gen_integer{17});
+        std::fill(output.begin(), output.end(), 0.0f);
+        std::fill(ref_out.begin(), ref_out.end(), 0.0f);
+        std::fill(bias.begin(), bias.end(), 0.0f);
+        int bias_mode = 1; // zero disables bias
+        convHostForward(input, ref_out, weights, bias_mode, bias, conv_desc);
+        activationHostInfer(cba_config.activ_mode,
+                            double_zero,
+                            double_zero,
+                            double_zero,
+                            ref_out.data,
```

Comment:
Should this be input.data instead?

Reply:
@atamazov @DrizztDoUrden could you take a look while @JehandadKhan is on vacation?